### ISSUE TITLE
Nuitka 2.1.5 causes segfault

### ISSUE THREAD

**Issue by sullivan50909** [Nuitka/Nuitka#3210]
- Please check whether the bug was already reported or fixed.
There are a few other tickets mentioning segfault. This may be related.

- Nuitka version, full Python version, flavor, OS, etc. as output by *this exact* command.
working version:
```
# python3 -m nuitka --version
2.4.9
Commercial: None
Python: 3.9.19 (main, Sep 11 2024, 00:00:00)
Flavor: Fedora Python
Executable: /usr/bin/python3
OS: Linux
Arch: x86_64
Distribution: Rhel (based on Fedora) 9.5
Version C compiler: /usr/bin/gcc (gcc 11).
```
Many versions before 2.4.9 worked. 2.4.11 also works. 

non-working version:
```
# python3 -m nuitka --version
2.5.1
Commercial: None
Python: 3.9.19 (main, Sep 11 2024, 00:00:00)
Flavor: Fedora Python
Executable: /usr/bin/python3
OS: Linux
Arch: x86_64
Distribution: Rhel (based on Fedora) 9.5
Version C compiler: /usr/bin/gcc (gcc 11).
```

- How did you install Nuitka and Python

`  pip3 install  nuitka`

- The specific PyPI names and versions

The list of pypi packages is the same between the working and non-working versions. 

- Also supply a Short, Self Contained, Correct, Example

  This will take me a few days to put together. I'll post an update in the comments.

The nuitka build process seems OK, unless I use the --debug flag. The segfault occurs when attempting to run my compiled code. 

```
# ./api.dist/api.bin
Nuitka: A segmentation fault has occurred. This is highly unusual and can
have multiple reasons. Please check https://nuitka.net/info/segfault.html
for solutions.
Segmentation fault (core dumped)
```

- Provide in your issue the Nuitka options used

  python3 -m nuitka --follow-imports --onefile --python-flag=no_docstrings api.py

  - Note if this is a regression

YES
Problem exists in 2.5.1. Problem does not exist in 2.4.11 or anything earlier that I've tested.

**Comment by sullivan50909** [comment]
This was actually pretty easy to reproduce. I did my test in a new ubi9 docker container.

create test.py with the following content:
print("Hello World")

yum -y install gcc pip python3-devel
pip3 install nuitka 
pip3 install patchelf 

python3 -m nuitka --onefile  test.py
./test.dist/test.bin

result:
Nuitka: A segmentation fault has occurred. This is highly unusual and can
have multiple reasons. Please check https://nuitka.net/info/segfault.html
for solutions.
Segmentation fault (core dumped)

Note that ./test.bin results in the expected Hello World output.

Now repeat with old Nuitka:
pip3 install nuitka==2.4.9
python3 -m nuitka --onefile  test.py
./test.dist/test.bin 

result: Hello World

**Comment by sullivan50909** [comment]
I don't think this is relevant, but you requested this in the form above. The only difference between working and non-working is the Nuitka version.
```
# pip list
Package              Version
-------------------- --------
chardet              4.0.0
dbus-python          1.2.18
decorator            4.4.2
gpg                  1.15.1
idna                 2.10
iniparse             0.4
libcomps             0.1.18
Nuitka               2.4.9
ordered-set          4.1.0
patchelf             0.17.2.1
pip                  21.3.1
PyGObject            3.40.1
pyinotify            0.9.6
PySocks              1.7.1
python-dateutil      2.8.1
requests             2.25.1
rpm                  4.16.1.3
setuptools           53.0.0
six                  1.15.0
subscription-manager 1.29.42
systemd-python       234
urllib3              1.26.5
zstandard            0.23.0
```

**Comment by kayhayen** [comment]
I updated the page to tell you what to do when `--debug` gives that result.

**Comment by kayhayen** [comment]
Since my CI has become weak and it might take a while to appear: --experimental=allow-c-warnings is what you can do next there

**Comment by kayhayen** [comment]
Given a docker file I could actually repeat it pretty easily, what is ubi9 ?

**Comment by sullivan50909** [comment]
ubi9 is a container image available here: https://hub.docker.com/r/redhat/ubi9
or `docker pull redhat/ubi9`
or if you are running podman, `podman pull ubi9`

**Comment by kayhayen** [comment]
That's very interesting, the `original_argv0` ought to not be possibly be None, likely a bug of Nuitka there, that somehow doesn't show everywhere, but with a Fedora Python it seems.

**Comment by sullivan50909** [comment]
The redhat ubi9 images are public. (Red Hat does offer paid products, but the ubi containers are free.) Are you running docker or podman or another container program?

**Comment by kayhayen** [comment]
> Error: creating build container: initializing source docker://redhat:ubi9: reading manifest ubi9 in docker.io/library/redhat: requested access to the resource is denied

This is podman, and I am using `bin/run-inside-nuitka-container` with a container file that tries to use `FROM redhat:ubi9` and gets that access error.

**Comment by kayhayen** [comment]
I checked with CentOS9:stream9 image, and no issue there.

**Comment by sullivan50909** [comment]
If you are using podman, try `podman run -it ubi9 bash` or `podman pull ubi9`

**Comment by sullivan50909** [comment]
I can't find a centos9:stream9 image.
I tested with docker.io/library/rockylinux:9 and got the same results described above.
My previous testing was done with registry.access.redhat.com/ubi9:latest

**Comment by kayhayen** [comment]
I am having severe skill issues it seems.

What is the equivalent of this for `ubi9` ?

```
RUN dnf install -y epel-release \
    && dnf repolist \
    && dnf install -y gcc strace patchelf ccache gdb make python3-devel python3-zstandard python3-ordered-set \
    && dnf clean -y all
```

The `FROM centos:stream9` worked for me, but don't switch or something, I would love to figure out how to make ubi9 work with you.

**Comment by sullivan50909** [comment]
What repos are available in ubi depends on your host system. Try 'dnf repolist' to make sure you have some repos available.

**Comment by kayhayen** [comment]
Thanks, that works for me; running the binary on the current development version gave me no issue, and I also thought it ought to be impossible for that to happen. I however had a though, where I think it might happen that multiprocessing children of a onefile binary would have to be able to run into this issue, since it's only set for the direct child of the onefile binary, and as such I am changing it to not handle onefile specific, but have that set properly for first child of a onefile, and wrongly for all others, but it's also not intended for them to be used.

I am using a container file like this

```
ARG CENTOS_VERSION
FROM ubi9
ENV container docker
USER root

RUN dnf install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-9.noarch.rpm \
    && dnf install -y epel-release \
    && dnf install -y gcc strace patchelf ccache gdb make python3-devel python3-zstandard python3-ordered-set \
    && dnf clean -y all

VOLUME [ "/sys/fs/cgroup" ]
CMD ["/usr/sbin/init"]
```

And then this works for me on my Windows:

```
python3.10 bin/run-inside-nuitka-container --container=./UBI9.containerfile --command="python3 bin/nuitka-run --debug tests/onefile/HelloWorldTest.py"
```

With both my current factory version and the develop version. Turns out, paths that are not templates do not work, so you might have to append a `.in` suffix, something I am only fixing with the next hotfix.

This says to me:

```
Nuitka: Launching 'HelloWorldTest.bin'.
Hello Onefile World!
Finished container run with exit code '0'.
```

And it never segfaults or asserts, so maybe your report is somehow incomplete or what.

**Comment by sullivan50909** [comment]
The problem appears to be resolved in branch: hotfix/2.5.2 commit: e923f127b598161
Thanks.

**Comment by kayhayen** [comment]
I have never seen any reproduction of the issue, closing for lack of feedback, I think I never had the actual reproducer, but at least I learned about ubi9 and how to use it too.

**Comment by sullivan50909** [comment]
The problem is resolved in Nuitka-2.5.4.
Thanks.
